src/test/scala/com/wixpress/common/specs2/ByNameArgumentsTest.scala

```diff
@@ -0,0 +1,42 @@
+package com.wixpress.common.specs2
+
+import org.specs2.mutable.Specification
+
+class ByNameArgumentsTest extends Specification with JMockTestSupport {
+  "support by name parameters" in withJMock { jmock =>
+    import jmock.{Stubbed, mock, allowing, checking}
+    val mocked = mock[ByNameArgs]
+    checking {
+      allowing(mocked).doSomething(2, "foo") willReturn 42
+    }
+
+    mocked.doSomething(1 + 1, "foo") === 42
+  }
+  "support Function0 parameters" in withJMock { jmock =>
+    import jmock.{Stubbed, mock, allowing, checking}
+    val mocked = mock[ThunkArgs]
+    checking {
+      allowing(mocked).doSomething(() => 2, "foo") willReturn 42
+    }
+
+    mocked.doSomething(() => 1 + 1, "foo") === 42
```

Comment:
this is great stuff! Just to make sure, matcher will still work here?

Reply:
Yes, but the matcher will be for the value: `doSomething(having(beEqualTo(2)), "foo")`

Reply:
hhm, I think it is a breaking change, right?

Reply:
I mean would `having(any[...])`work?

Reply:
Yes, `having(any[...])` any will work. 
And given that this is the only thing that could work before, I think this change shouldn't break anyone